### PR TITLE
Tests Collection: add always-collected packs

### PR DESCRIPTION
**Pull request by dorschw**
- Add `Base` and `DeveloperTools` to always be collected
- `Whois` is now collected when the sanity test based on it is collected
- improve logging clarity for skipped collections (e.g. unit tests change)
- remove exception raised when no files are to be collected (this is no longer considered an error, since files were separated `triggering-sanity-test collection` and `not-triggering-sanity-test collection` #20393)
- Fixed a bug where integrations missing from conf.json [caused](https://code.pan.run/xsoar/content/-/jobs/15340019#L2898) a `KeyError`, now logging a warning instead.
- Added `DictBased._tests` and `DictBased.explicit_no_tests` for storing the actual value from self['tests'], and checking whether `no tests` is specified. The `Dictbased.tests` behavior is unchanged - raising when necessary.

### CODE REVIEW

File: Tests/scripts/collect_tests/collect_tests.py [hunk 1]
```diff
@@ -102,7 +106,8 @@ def _validate_args(pack: Optional[str], test: Optional[str], reason: CollectionR
         """
         Validates the arguments of the constructor.
         """
-        if reason != CollectionReason.DUMMY_OBJECT_FOR_COMBINING:
+        if reason not in {CollectionReason.DUMMY_OBJECT_FOR_COMBINING,
+                          CollectionReason.ALWAYS_INSTALLED_PACKS}:
             for (arg, arg_name) in ((conf, 'conf.json'), (id_set, 'id_set')):
                 if not arg:
                     # may be None only when reason==DUMMY_OBJECT_FOR_COMBINING
```
Comment (yuvalbenshalom) [inline Tests/scripts/collect_tests/collect_tests.py:113]: maybe you should update this comment to ALWAYS_INSTALLED_PACKS also.

File: Tests/scripts/collect_tests/collect_tests.py [hunk 2]
```diff
@@ -438,14 +471,22 @@ def _get_changed_files(self) -> tuple[str, ...]:
...
         if contrib_diff:
             logger.debug('adding contrib_diff to diff')
             diff = f'{diff}\n{contrib_diff}'
 
         # diff is formatted as `M  foo.json\n A  bar.py\n ...`, turning it into ('foo.json', 'bar.py', ...).
-        return tuple((value.split()[1] for value in filter(None, diff.split('\n'))))
+        files = []
+        for line in filter(None, diff.splitlines()):
+            status, path = line.split()
+            if status != 'D':
```
Comment (yuvalbenshalom) [inline Tests/scripts/collect_tests/collect_tests.py:484]: status D?

**Review by yuvalbenshalom — COMMENTED** [review]
looks good.
see my comments.

File: Tests/scripts/collect_tests/collect_tests.py (hunk 2 above)
Comment (dorschw) [reply Tests/scripts/collect_tests/collect_tests.py:484]: deleted, explained in the following rows, i'll move the comment here for clarity.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**